scripts/submit_bb.py

```diff
@@ -36,129 +37,178 @@ def gen_command_template(params):
     return command_template_parameters, params.bb
 
 
-def main(args: argparse.Namespace, logger: Logger = get_basic_logger()):
-    params = utils.load_sim_params(os.path.join(args.rel_dir, "sim_params.yaml"))
-    ncores = platform_config[const.PLATFORM_CONFIG.BB_DEFAULT_NCORES.name]
+def main(
+    auto: bool = False,
+    machine: str = host,
+    ncores: int = platform_config[const.PLATFORM_CONFIG.BB_DEFAULT_NCORES.name],
+    rel_dir: Path = Path("."),
+    retries: int = 0,
+    version: str = None,
+    write_directory: Path = None,
+    logger: Logger = get_basic_logger(),
+):
+    rel_dir = rel_dir.resolve()
+    try:
+        params = utils.load_sim_params(rel_dir / "sim_params.yaml")
```

Comment:
how does this string operation work? is it missing something? I never seen this before.

Reply:
https://realpython.com/python-pathlib/

Reply:
oh i see now, i did not realized that you changed the type to from basic os.path string to pathlib path.